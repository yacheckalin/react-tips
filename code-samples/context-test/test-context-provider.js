import React, { useState, useContext } from "react";

export const TestContext = React.createContext();
export const useTestContext = () => useContext(TextContext);

const TestContextProvider = ({ children }) => {
  const [data, setData] = useState([]);

  return (
    <TestContext.Provider value={{ data, setData }}>
      {children}
    </TestContext.Provider>
  );
};

export default TestContextProvider;
