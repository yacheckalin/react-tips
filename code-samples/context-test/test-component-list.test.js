import React from "react";
import toJSON from "enzyme-to-json";
import { shallow } from "enzyme";

import TestComponentList from "./test-component-list";
import * as TestContext from "./test-context-provider";

describe("<TestComponentList />", () => {
  test("shows and displays properly", () => {
    const contextValues = { setData: jest.fn(), data: [] };
    jest
      .spyOn(TestContext, "useTestContext")
      .mockImplementation(() => contextValues);
    const wrapper = shallow(<TestComponentList />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
