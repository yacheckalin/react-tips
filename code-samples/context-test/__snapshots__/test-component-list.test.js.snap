// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TestComponentList /> shows and displays properly 1`] = `
<div
  className="col s4"
>
  <ul
    className="collection"
  />
</div>
`;
